packages/core/ui-settings/core-ui-settings-browser-internal/src/ui_settings_api.ts

```diff
@@ -94,6 +98,31 @@ export class UiSettingsApi {
     });
   }
 
+  /**
+   * Sends a validation request to the server for the provided key+value pair.
+   */
+  public validate(key: string, value: any) {
+    return new Promise<ValidationApiResponse>((resolve, reject) => {
+      try {
+        const path = `/internal/kibana/settings/${key}/validate`;
+        const resp = this.sendRequest('POST', path, {
+          value,
+        });
+        resolve(resp);
+      } catch (error) {
+        if (error.response) {
+          if (error.response.status === 400) {
+            throw new Error(error.body.message);
+          }
+          if (error.response.status > 400) {
+            throw new Error(`Request failed with status code: ${error.response.status}`);
+          }
+        }
+        reject(error);
+      }
+    });
+  }
```

Comment:
Yes, indeed this looks better. In fact, since we are anyway catching any errors in [validateValue()](https://github.com/elastic/kibana/blob/4f5fbcb2d5e14ab7f5fe3cfe6f0154bdaf436ca7/packages/core/ui-settings/core-ui-settings-browser-internal/src/ui_settings_client_common.ts#L138C7-L138C7), would it be wrong if we simply do it as below?
```ts
public async validate(key: string, value: any): Promise<ValidationApiResponse> {
    return await this.sendRequest('POST', `/internal/kibana/settings/${key}/validate`, {
      value,
    });
}
```